backend/src/routes/UserRoutes.ts

```diff
@@ -14,6 +22,12 @@ if (config.nodeEnv === 'development') {
 UserRouter.post('/login', loginUser);
 
 UserRouter.route('/').get(protect(Permission.VIEW_USERS), getAllUsers);
-UserRouter.route('/:id').patch(protect(Permission.UPDATE_USERS), updateUser);
+UserRouter.route('/:userId')
+    .get(protect(Permission.VIEW_USERS), getUser)
+    .patch(protect(Permission.UPDATE_USERS), updateUser);
+
+UserRouter.route('/:userId/roles/')
+    .post(protect(Permission.GIVE_ROLE), giveRoles)
+    .delete(protect(Permission.REMOVE_ROLE), removeRoles);
```

Comment:
I think instead of creating more permissions, these should reuse `Permission.UPDATE_USERS` as this can be done with that endpoint anyway, this is just a convenience function, what do you think though?

Reply:
I think next sprint I'll focus on adding in functionality with role hierarchy as discussed in the discord. 

I'll improve the permissions with those changes however I believe that `Permission.UPDATE_USERS` will be separate from say `Permission.MANAGE_ROLES` because update users will probably be a very admin type permission once the user schema starts containing wellbeing data etc and other things we don't want just anyone to be able to update.

Reply:
Actually yeah, that's a good point